src/server/scripts/Northrend/VioletHold/boss_moragg.cpp

```diff
@@ -134,7 +147,149 @@ class boss_moragg : public CreatureScript
     }
 };
 
+class spell_moragg_ray_of_suffering : public SpellScriptLoader
+{
+public:
+    spell_moragg_ray_of_suffering() : SpellScriptLoader("spell_moragg_ray_of_suffering") { }
+
+    class spell_moragg_ray_of_suffering_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_moragg_ray_of_suffering_AuraScript);
+
+        void OnPeriodic(AuraEffect const* aurEff)
+        {
+            PreventDefaultAction();
+            std::list<HostileReference*> players = GetTarget()->ToCreature()->getThreatManager().getThreatList();
+            if (!players.empty())
+            {
+                std::list<HostileReference*>::iterator itr = players.begin();
+                std::advance(itr, urand(0, players.size() - 1));
+
+                uint32 triggerSpell = GetSpellInfo()->Effects[aurEff->GetEffIndex()].TriggerSpell;
+                GetTarget()->CastCustomSpell(triggerSpell, SPELLVALUE_MAX_TARGETS, 1, (*itr)->getTarget(), TRIGGERED_FULL_MASK, NULL, aurEff);
+            }
+        }
+
+        void Register() override
+        {
+            OnEffectPeriodic += AuraEffectPeriodicFn(spell_moragg_ray_of_suffering_AuraScript::OnPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_TRIGGER_SPELL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_moragg_ray_of_suffering_AuraScript();
+    }
+};
+
+class spell_moragg_ray_of_pain : public SpellScriptLoader
+{
+public:
+    spell_moragg_ray_of_pain() : SpellScriptLoader("spell_moragg_ray_of_pain") { }
+
+    class spell_moragg_ray_of_pain_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_moragg_ray_of_pain_AuraScript);
+
+        void OnPeriodic(AuraEffect const* aurEff)
+        {
+            PreventDefaultAction();
+            std::list<HostileReference*> players = GetTarget()->ToCreature()->getThreatManager().getThreatList();
```

Comment:
indeed @oMadMano the ToCreature() cast looks totally useless since getThreatManager() is in Unit class, well spotted